src/components/Microsoft.DotNet.UpgradeAssistant/UpgradeResultCollector.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    internal class UpgradeResultCollector : IUpgradeResultCollector
```

Comment:
```suggestion
    internal class UpgradeResultCollector : IUpgradeResultCollector, IEnumerable<AnalyzeResultDefinition>
```

and implement `IEnumerable<T>` via `_resultCollection`